runtime/marshal_jsonpb.go

```diff
@@ -30,10 +30,6 @@ func (*JSONPb) ContentType(_ interface{}) string {
 
 // Marshal marshals "v" into JSON.
 func (j *JSONPb) Marshal(v interface{}) ([]byte, error) {
-	if _, ok := v.(proto.Message); !ok {
-		return j.marshalNonProtoField(v)
-	}
```

Comment:
I'm a bit worried about removing this, why was it necessary?

Reply:
This appears to be unused code.  After that type checking calls j.marshalTo, where the same type checking is performed and the same call of j.marshalNonProtoField if the interface type is not proto.Message.

After removing the j.marshalNonProtoField call from j.Marshal, we can use json.Indent to adjust the indentation after the j.marshalNonProtoField call in one place. Otherwise we need to use json.Indent after j.marshalNonProtoFields in j.Marshal too.

Reply:
Ah, fair, thank you.